apps/prairielearn/src/components/QuestionScore.html.ts

```diff
@@ -249,12 +249,33 @@ export function ExamQuestionScore({
 
 export function ExamQuestionStatus({
   instance_question,
+  assessment_question,
 }: {
   instance_question: InstanceQuestion & {
     allow_grade_left_ms?: number;
     allow_grade_interval?: string;
   };
+  assessment_question: Pick<AssessmentQuestion, 'max_auto_points' | 'max_manual_points'>;
 }) {
+  // Special case: if this is a manually graded question in the "saved" state,
+  // we want to differentiate it from saved auto-graded questions which can
+  // be graded immediately. We'll use a green badge so that student can drive
+  // towards all status badges being green.
+  //
+  // TODO: can we safely look at the assessment question for exams? What about
+  // the guarantee that an Exam-type assessment won't change after it's created?
```

Comment:
👀 I'd appreciate some help thinking through this case.

Reply:
Referring to assessment questions is definitely a problem, but we already do it elsewhere so I don't think we are making things any worse here.

I've been thinking for a long time that we need to improve our support for modifying in-progress assessments (especially exams). I need to write this up, but the key thing is that we should make a per-assessment-instance copy of zones and assessment questions at the time that the assessment instance is created. Ideally we'd also lock question versions once question versioning is implemented. This then allows the assessment instance to be unaffected by any changes to the original assessment. We'd then need to add explicit operations to "reset" an assessment instance to the latest version of the assessment. I haven't yet thought through how best to handle the zone/assessment question duplication with the DB -- separate tables, a flag in the existing tables, etc? It's also not clear to me whether we should also be duplicating some/all of the information in the assessments table as well.